src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplate.java

```diff
@@ -653,9 +659,14 @@ public Boolean doInBucket() throws InterruptedException, ExecutionException {
     execute(new BucketCallback<Boolean>() {
       @Override
       public Boolean doInBucket() {
-        RawJsonDocument deletedDoc = client.remove(converted.getId(), persistTo, replicateTo
-            , RawJsonDocument.class);
-        return deletedDoc != null;
+        try {
+          RawJsonDocument deletedDoc = client.remove((String) objectToRemove, persistTo, replicateTo,
```

Comment:
this seems copy-paste from row 643 above. anyway it is wrong, and the integration tests would have shown that (compare again to the code from before the change).